include/aspect/introspection.h

```diff
@@ -294,6 +294,28 @@ namespace aspect
       IndexSets index_sets;
 
       /**
+       * A struct that describes the available methods for
+       * compositional fields. This type is used to determine how
+       * to advect a compositional field.
+       */
+      struct FieldMethod
```

Comment:
Is this going to be used for anything other than the compositional fields? `FieldMethod` is not very descriptive. Maybe call it `CompositionalFieldComputation`? I don't particularly care for that name either, so if you can come up with something better, go with that!